pkg/kubelet/container/runtime_test.go

```diff
@@ -0,0 +1,378 @@
+	assert.Equal(t, networkReadyCondition, actualForNetworkReady)
+}
+
+func TestFindPodByID(t *testing.T) {
+	expect := Pod{
+		ID:   "id-pod2",
+		Name: "pod2",
+	}
+
+	expectNoResult := Pod{}
+
+	pods := Pods{
+		{
+			ID:   "id-pod1",
+			Name: "pod1",
+		},
+		{
+			ID:   "id-pod2",
+			Name: "pod2",
+		},
+	}
+
+	actual := pods.FindPodByID(expect.ID)
+	assert.Equal(t, expect, actual)
+
+	actualNoResult := pods.FindPodByID("id-pod3")
+	assert.Equal(t, actualNoResult, expectNoResult)
+
+}
+
+func TestFindPodByFullName(t *testing.T) {
+	expect := Pod{
+		Namespace: "ns2",
+		Name:      "pod2",
+	}
+
+	expectNoResult := Pod{}
+
+	pods := Pods{
+		{
+			Namespace: "ns1",
+			Name:      "pod1",
+		},
+		{
+			Namespace: "ns2",
+			Name:      "pod2",
+		},
+	}
+
+	actual := pods.FindPodByFullName("pod2_ns2")
+	assert.Equal(t, expect, actual)
+
+	actualNoResult := pods.FindPodByFullName("pod3_ns3")
+	assert.Equal(t, actualNoResult, expectNoResult)
+}
+
+func TestFindContainerByName(t *testing.T) {
+	expect := &Container{
+		Name: "container1",
+	}
+
+	pod := &Pod{
+		Name: "pod1",
+		Containers: []*Container{
+			{
+				Name: "container1",
+			},
+			{
+				Name: "container2",
+			},
+		},
+	}
+
+	actual := pod.FindContainerByName("container1")
+	assert.Equal(t, expect, actual)
+
+	actualNoResult := pod.FindContainerByName("container3")
+	assert.Nil(t, actualNoResult)
+}
+
+func TestFindContainerByID(t *testing.T) {
+	expect := &Container{
+		Name: "container1",
+		ID: ContainerID{
+			Type: "docker",
+			ID:   "1111",
+		},
+	}
+
+	pod := &Pod{
+		Name: "pod1",
+		Containers: []*Container{
+			{
+				Name: "container1",
+				ID: ContainerID{
+					Type: "docker",
+					ID:   "1111",
+				},
+			},
+			{
+				Name: "container2",
+				ID: ContainerID{
+					Type: "docker",
+					ID:   "2222",
+				},
+			},
+		},
+	}
+
+	actual := pod.FindContainerByID(ContainerID{
+		Type: "docker",
+		ID:   "1111",
+	})
+	assert.Equal(t, expect, actual)
+
+	actualNoResult := pod.FindContainerByID(ContainerID{
+		Type: "docker",
+		ID:   "3333",
+	})
+	assert.Nil(t, actualNoResult)
+}
+
+func TestFindSandboxByID(t *testing.T) {
+	expect := &Container{
+		Name: "container1",
+		ID: ContainerID{
+			Type: "docker",
+			ID:   "1111",
+		},
+	}
+
+	pod := &Pod{
+		Name: "pod1",
+		Sandboxes: []*Container{
+			{
+				Name: "container1",
+				ID: ContainerID{
+					Type: "docker",
+					ID:   "1111",
+				},
+			},
+			{
+				Name: "container2",
+				ID: ContainerID{
+					Type: "docker",
+					ID:   "2222",
+				},
+			},
+		},
+	}
+
+	actual := pod.FindSandboxByID(ContainerID{
+		Type: "docker",
+		ID:   "1111",
+	})
+	assert.Equal(t, expect, actual)
+
+	actualNoResult := pod.FindSandboxByID(ContainerID{
+		Type: "docker",
+		ID:   "3333",
+	})
+	assert.Nil(t, actualNoResult)
+}
+
+func TestParsePodFullName(t *testing.T) {
+	fullNameWithoutLine := "name3ns3"
+	testCases := []struct {
+		expectName      string
+		expectNmaespace string
```

Comment:
Throughout func fix typo:

```suggestion
		expectNamespace string
```

Reply:
thanks!